infrastructure/iam.tf

```diff
@@ -45,6 +45,26 @@ module "iam_iam-policy" {
                 "ecr:DescribeImageScanFindings"
             ],
             "Resource": "*"
+        },
+        {
+            "Effect": "Allow",
+            "Action": [
+                "eks:AccessKubernetesApi",
```

Comment:
This is to much, try to follow **principle of least privilege**. 
GitHub should just be able to create eks config and update image tag on the pod level.
Second part is controlled via **access_entries** which you already defined in eks module.